Add method on Material for getting number of atoms

paulromano opened the pull request:
This PR adds a new `Material.get_nuclide_atoms` method that returns a dictionary listing the number of atoms of each nuclide in the material. Along with that, it also changes the return value of the `get_nuclide_atom_densities`, which before would return a dictionary that looks like:
```Python
{'U234': ('U234', 4.989e-6),
 'U235': ('U235', 5.581e-4),
 ...}
```
As you can see, the values in the dictionary would superfluously store the nuclide name. Now, that method will return a dictionary that looks like
```Python
{'U234': 4.989e-6,
 'U235': 5.581e-4,
 ...}
```

Finally, there's an unrelated small fix in here to catch a divide-by-zero when calculating the `decay_constant` property of the `Decay` class when an ENDF decay evaluation lists a zero half-life.

shimwell commented:
This is a nice improvement, I guess the next step is to add a ```get_nuclide_activity```. Something that just returns this dictionary with the activity of each nuclide in Bq. I would be happy to add that if that feature is required elsewhere? Perhaps handy for shut down dose rate calcs ? @eepeterson @paulromano

paulromano commented:
@shimwell I just added the `get_nuclide_activity` method as you suggested. It also seemed to make sense to have a separate `openmc.data.decay_constant` function that returns ln(2) / half_life so I added that in as well.

shimwell commented:
Happy to merge this one as I'm quite familiar with the material class and this functionality, all the tests pass and everything looks good.
